src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -1430,6 +1431,15 @@ protected boolean _isIgnorable(Annotated a)
         if (ann != null) {
             return ann.value();
         }
+        JsonIgnoreIf jsonIgnoreIfAnn = _findAnnotation(a, JsonIgnoreIf.class);
+        if(jsonIgnoreIfAnn != null) {
+            try {
+                JsonIgnoreValidator jsonIgnoreValidator = jsonIgnoreIfAnn.value().newInstance();
```

Comment:
This has to be changed, handlers would need to be instantiated using `HandlerInstantiator` as is done with other helper objects. Alas, method is not yet passed `MapperConfig` and it's not trivial to get it there...